### PR TITLE
Hiding Other Requirements doesn't prevent course from being published in 2025

### PR DESCRIPTION
**Pull request by inulty-dfe**
### Context

We've added a route constraint to prevent some endpoints being accessible for the 2025 recruitment cycle.

However, these named routes are still evaluated even if they are not executed in the view.

### Changes proposed in this pull request

_Remove the Feature Flag for course requirements deprecation. It's not necessary becuase of the separation of courses by recruitment cycle._

Wrap the named routes in a block that executes conditionally to prevent routing error.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Review by elceebee — APPROVED** [review]

File: app/models/provider.rb [hunk 1]
```diff
@@ -382,7 +382,7 @@ def tda_active?
   end
 
   def course_requirements_deprecated?
-    recruitment_cycle.after_2024? || FeatureService.enabled?(:course_requirements_deprecated)
+    recruitment_cycle.after_2024?
```
[the remark this answers is not in the document]
Comment (inulty-dfe) [reply app/models/provider.rb:385]: I'm not totally familiar with the plans you've described.

This code just prevents them appearing on 2025 courses. They will still be visible on 2024 courses as long as 2024 courses are visible, which is not a long time. I think this is less effort and less disruptive if we just let this die with the 2024 recruitment cycle.

This code can be removed when we enter the new recruitment cycle.

**Comment by inulty-dfe** [comment]
[Investigate not being able to publish courses in the next recruitment cycle](https://trello.com/c/Puos6yEr/1771-investigate-not-being-able-to-publish-courses-in-the-next-recruitment-cycle)

**Review by avinhurry — APPROVED** [review]
This is good, a couple of suggestions/comments in line.

My main concern is the timing that we want these fields to be hidden, but it sounds like we only want to surface this in the next cycle.

File: spec/features/publish/courses/publishing_a_course_2025_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (inulty-dfe) [reply spec/features/publish/courses/publishing_a_course_2025_spec.rb:1]: I feel the fact it has 2025 in it's filename is an indication of it's limited value

File: app/helpers/publish_helper.rb [hunk 3]
```diff
@@ -40,4 +40,8 @@ def ruby_pants_options
       en_dash: '–'
     }
   end
+
+  def render_if(condition, &)
```
[the remark this answers is not in the document]
Comment (inulty-dfe) [reply app/helpers/publish_helper.rb:44]: The block is the essential ingredient in this fix. The only other way I can think is not pretty:

```
<% condition && proc do %>
<% end.call %>
```

[the remark this answers is not in the document]
Comment (inulty-dfe) [reply app/helpers/publish_helper.rb:44]: The block is lazy evaluated. The bug is that the template evaluates the named route when it doesn't exist for 2025

File: spec/features/publish/courses/publishing_a_course_2025_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (inulty-dfe) [reply spec/features/publish/courses/publishing_a_course_2025_spec.rb:1]: I've made the changes suggested to this file

File: app/helpers/publish_helper.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (avinhurry) [reply app/helpers/publish_helper.rb:44]: Oh right, that makes sense. Sorry.

**Pull request merged**